packages/react-bindings/test/hooks/useAutoControlled-test.tsx

```diff
@@ -0,0 +1,93 @@
+import { useAutoControlled } from '@fluentui/react-bindings'
+import { shallow } from 'enzyme'
+import * as React from 'react'
+import * as ReactTestUtils from 'react-dom/test-utils'
+
+type TestComponentProps = {
+  defaultValue?: string
+  value?: string
+
+  initialValue?: string
+  onChange?: (value: string) => void
+}
+
+const TestComponent: React.FunctionComponent<TestComponentProps> = props => {
+  const [value, setValue] = useAutoControlled({
+    defaultValue: props.defaultValue,
+    value: props.value,
+
+    initialValue: props.initialValue,
+  })
+
+  return (
+    <>
+      <input
+        onChange={e => {
+          setValue(e.target.value)
+          if (props.onChange) props.onChange(e.target.value)
+        }}
+        value={value || ''}
+      />
+      <button onClick={() => setValue(undefined)} />
+    </>
+  )
+}
+
+describe('useAutoControlled', () => {
+  it('defaults to "undefined"', () => {
+    const wrapper = shallow(<TestComponent />)
+
+    expect(wrapper.find('input').prop('value')).toBe('')
+  })
+
+  it('defaults to `initialValue` if specified', () => {
+    const wrapper = shallow(<TestComponent initialValue="foo" />)
+
+    expect(wrapper.find('input').prop('value')).toBe('foo')
+  })
+
+  it('sets default values of controlled props to the state', () => {
+    const wrapper = shallow(<TestComponent defaultValue="foo" />)
+
+    expect(wrapper.find('input').prop('value')).toBe('foo')
+  })
+
+  it('sets values of controlled props to the state', () => {
+    const wrapper = shallow(<TestComponent value="foo" />)
+
+    expect(wrapper.find('input').prop('value')).toBe('foo')
+  })
+
+  it('handles state updates via actions', () => {
+    const wrapper = shallow(<TestComponent />)
```

Comment:
I would also add this test for `<TestComponent value="foo" />` and `<TestComponent defaultValue="foo" />`